src/model_config_tests/qa/test_access_esm1p5_config.py

```diff
@@ -0,0 +1,167 @@
+# Copyright 2024 ACCESS-NRI and contributors. See the top-level COPYRIGHT file for details.
+# SPDX-License-Identifier: Apache-2.0
+
+"""ACCESS-ESM1.5 specific configuration tests"""
+
+import re
+import warnings
+from typing import Any
+
+import pytest
+
+from model_config_tests.util import get_git_branch_name
+
+### Bunch of expected values for tests ###
+VALID_REALMS: set[str] = {"atmosphere", "land", "ocean", "ocnBgchm", "seaIce"}
+VALID_KEYWORDS: set[str] = {"global", "access-esm1.5"}
+VALID_NOMINAL_RESOLUTION: str = "100 km"
+VALID_REFERENCE: str = "https://doi.org/10.1071/ES19035"
+VALID_PREINDUSTRIAL_START: dict[str, int] = {"year": 101, "month": 1, "day": 1}
+VALID_HISTORICAL_START: dict[str, int] = {"year": 1850, "month": 1, "day": 1}
+VALID_RUNTIME: dict[str, int] = {"year": 1, "month": 0, "day": 0}
+VALID_RESTART_FREQ: str = "20YS"
+
+
+### Some functions to avoid copying assertion error text
+def error_field_nonexistence(field: str, file: str) -> str:
+    return f"Field '{field}' is null or does not exist in {file}."
+
+
+def error_field_incorrect(field: str, file: str, expected: Any) -> str:
+    return f"Field '{field}' in {file} is not expected value: {expected}"
+
+
+class AccessEsm1p5Branch:
+    """Use the naming patterns of the branch name to infer information of
+    the ACCESS-ESM1.5 config"""
+
+    def __init__(self, branch_name):
+        self.branch_name = branch_name
+        self.config_scenario = self.set_config_scenario()
+        self.config_modifiers = self.set_config_modifiers()
+
+    def set_config_scenario(self) -> str:
+        # Regex below is split into three sections:
+        # Config type start section: '^.+-' for 'release-', 'dev-'...
+        # Scenario section: '([^+]+)' for 'preindustrial', 'historical'...anything that isn't the '+' modifier sigil
+        # Modifiers end section: '(?:\+.+)*' any amount of '+modifer' sections
+        scenario_match = re.match(
+            r"^.+-(?P<scenario>[^+]+)(?:\+.+)*$", self.branch_name
```

Comment:
Should this be more explicit and require either `release` or `dev` before the `-` separator? I kinda think we should. We're not promising to deal with anything other than that are we?

Also do we care if it is `release` or `dev`? I guess not if we're not saving it in the state, just curious.

Reply:
Good shout on the regex. Will update. 
I can't think of any reason for QA tests to treat them any differently whether they're `release-*` or `dev-*`, so I don't think we need to capture that information.

Reply:
Updated in https://github.com/ACCESS-NRI/model-config-tests/pull/41/commits/c4d8a7d130b1a3764c34ba3c7cd0cbd7cb7971f8

Reply:
And in https://github.com/ACCESS-NRI/model-config-tests/pull/41/commits/8f6ff1e88ec3f5872757e78867c964d0625c02d0 because I had a strange `?` after `(?:release|dev)`